### PR TITLE
Enable ORY Hydra Maester in Kyma

### PR DESCRIPTION
**Pull request by jakkab**
**Description**

Changes proposed in this pull request:

- plug Hydra Maester in
- docu

### CODE REVIEW

File: docs/security/03-06-oauth2-server.md [hunk 1]
```diff
@@ -3,31 +3,45 @@ title: OAuth2 and OpenID Connect server
...
+  grantTypes:
+    - "client_credentials"
+  scope: "read write"
+EOF
+```
+
+As a result, Hydra Maester sends a registration request to the OAuth2 server and saves incoming credentials to a Kubernetes Secret.
+>**CAUTION:** The instance of the`oauth2clients.hydra.ory.sh` custom resource and its corresponding Secret share a name and Namespace.
+
+To retrieve your client credentials granted by the OAuth2 sever, run this command:
```
Comment (tomekpapiernik) [inline docs/security/03-06-oauth2-server.md]:
```suggestion
Run this command to get the credentials of the registered OAuth2:
```

File: docs/security/03-06-oauth2-server.md [hunk 2]
```diff
@@ -3,31 +3,45 @@ title: OAuth2 and OpenID Connect server
...
+```
+kubectl get secret -n my-namespace my-oauth2-client -o yaml
```
Comment (tomekpapiernik) [inline docs/security/03-06-oauth2-server.md]:
```suggestion
kubectl get secret -n {CLIENT_NAMESPACE} {NAME_OF_CLIENT} -o yaml
```

File: docs/security/03-06-oauth2-server.md [hunk 3]
```diff
@@ -3,31 +3,45 @@ title: OAuth2 and OpenID Connect server
...
+```
+
+Visit [ORY Helm Charts](https://github.com/ory/k8s/blob/master/docs/helm/hydra-maester.md) Github page to learn more about the `oauth2clients.hydra.ory.sh` custom resource.
```
Comment (tomekpapiernik) [inline docs/security/03-06-oauth2-server.md]:
```suggestion
See [this](https://github.com/ory/k8s/blob/master/docs/helm/hydra-maester.md) ORY document to learn more about the `oauth2clients.hydra.ory.sh` custom resource.
```

File: docs/security/03-06-oauth2-server.md [hunk 4]
```diff
@@ -3,31 +3,45 @@ title: OAuth2 and OpenID Connect server
...
+
+Creating this custom resource triggers the Hydra Maester controller which sends a client registration request to the OAuth2 server and saves the credentials of the registered client to a Kubernetes Secret.
+>**NOTE:** The instance of the`oauth2clients.hydra.ory.sh` custom resource and the Secret that stores the credentials of the corresponding client share the name and the Namespace.
+
+Run this command to get the credentials of the registered OAuth2:
+```
+kubectl get secret -n {CLIENT_NAMESPACE} {NAME_OF_CLIENT} -o yaml
+```
+
+See Ory Hydra Maester [github page](https://github.com/ory/hydra-maester) to learn more about the `oauth2clients.hydra.ory.sh` custom resource.
```
Comment (tomekpapiernik) [inline docs/security/03-06-oauth2-server.md]:
```suggestion
See the ORY Hydra Maester [github page](https://github.com/ory/hydra-maester) to learn more about the `oauth2clients.hydra.ory.sh` custom resource.
```

**Comment by jakkab** [comment]
see #5500

**Pull request merged**